protocol_data/rhinofi.py

```diff
@@ -0,0 +1,687 @@
+import json
+
+PROTOCOL_NAME = "rhinofi"
+
+def get_contract_address(chain_id, type):
+    # Call by scraping logic to determine where to monitor for events
+    # source: https://github.com/rhinofi/contracts_public/
+    contracts = {
+
+        # ethereum is not clear, there is no bridge contract
+        # there is a deposit contract but no fill contract
+        # '1': {
+        #     'deposit': { 12062072 : '0xeD9d63a96c27f87B07115b56b2e3572827f21646' },
+        #     'fill': { STARTING_BLOCK_NUMBER : '' }
+        # },
+        '42161' : { 
+            'deposit': { 12062072 : '0x10417734001162Ea139e8b044DFe28DbB8B28ad0' },
+            'fill': { 12062072 : '0x10417734001162Ea139e8b044DFe28DbB8B28ad0' },
+        },
+        '56' : { 
+            'deposit': { 21550595 : '0xB80A582fa430645A043bB4f6135321ee01005fEf' },
+            'fill': { 21550595 : '0xB80A582fa430645A043bB4f6135321ee01005fEf' },
+        },
+        '137': {
+            'deposit': { 16917268 : '0xBA4EEE20F434bC3908A0B18DA496348657133A7E' },
+            'fill': { 16917268 : '0xBA4EEE20F434bC3908A0B18DA496348657133A7E' }
+        },
+        '324': {
+            'deposit': { 193 : '0x1fa66e2B38d0cC496ec51F81c3e05E6A6708986F' },
+            'fill': { 193 : '0x1fa66e2B38d0cC496ec51F81c3e05E6A6708986F' }
+        },
+        '1101': {
+            'deposit': { 15845 : '0x65A4b8A0927c7FD899aed24356BF83810f7b9A3f' },
+            'fill': { 15845 : '0x65A4b8A0927c7FD899aed24356BF83810f7b9A3f' }
+        },
+        '10': {
+            'deposit': { 96888536 : '0x0bCa65bf4b4c8803d2f0B49353ed57CAAF3d66Dc' },
+            'fill': { 96888536 : '0x0bCa65bf4b4c8803d2f0B49353ed57CAAF3d66Dc' }
+        },
+        '59144': {
+            'deposit': { 593 : '0xcF68a2721394dcf5dCF66F6265C1819720F24528' },
+            'fill': { 593 : '0xcF68a2721394dcf5dCF66F6265C1819720F24528' }
+        },
+        '8453': {
+            'deposit': { 1448656 : '0x2f59E9086ec8130E21BD052065a9E6B2497bb102' },
+            'fill': { 1448656 : '0x2f59E9086ec8130E21BD052065a9E6B2497bb102' }
+        },
+        '169': {
+            'deposit': { 21324 : '0x2B4553122D960CA98075028d68735cC6b15DeEB5' },
+            'fill': { 21324 : '0x2B4553122D960CA98075028d68735cC6b15DeEB5' }
+        },
+        '534352': {
+            'deposit': { 694 : '0x87627c7E586441EeF9eE3C28B66662e897513f33' },
+            'fill': { 694 : '0x87627c7E586441EeF9eE3C28B66662e897513f33' }
+        },
+    }
+    return contracts[chain_id][type]
+
+# TODO: find solana contracts, they are absent in https://github.com/rhinofi/contracts_public/
+def get_function_identifier(chain_id, type):
```

Comment:
I've asked about this, let's hold off until we get a response

Reply:
solana won't work anyways - we match based on address, solana addresses are not evm compatible